File: programmers/dfs&bfs/network.py
# https://programmers.co.kr/learn/courses/30/lessons/43162

def solution(n, computers):
    dj_set = [m for m in range(n)]
    for i, others in enumerate(computers):
        for j, is_connected in enumerate(others):
            if i == j or not is_connected:
                continue

            if dj_set[i] != dj_set[j]:
                if dj_set[i] > dj_set[j]:
                    dj_set[i] = dj_set[j]
                else:
                    dj_set[j] = dj_set[i]

    return len(set(dj_set))

File: programmers/dfs&bfs/network.cc
// https://programmers.co.kr/learn/courses/30/lessons/43162

#include <stack>
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<vector<int>> computers) {
    int n_computer = computers.size(), cnt = 0;
    vector<int> visited(n_computer, 0);
    stack<int> network;
    for (int i = 0; i < n_computer; i++) {
        if (!visited[i]) {
            network.push(i);
            visited[i] = 1;
            while (!network.empty()) {
                auto curr = network.top();
                network.pop();
                for (int j = 0; j < n_computer; j++) {
                    if (!visited[j] && computers[curr][j]) {
                        network.push(j);
                        visited[j] = 1;
                    }
                }
            }
            cnt++;
        }
    }
    return cnt;
}